ui/src/layout/signin/RootComponent.js

```diff
@@ -27,22 +27,17 @@ const AnimationItem = (props) => {
   )
 }
 
-@withTranslation()
-class TiDBSignInForm extends React.PureComponent {
-  state = {
-    loading: false,
-    signInError: null,
-  }
+function TiDBSignInForm({ registry }) {
+  const { t } = useTranslation()
+  const [loading, setLoading] = useState(false)
+  const [signInError, setSignInError] = useState(null)
 
-  constructor(props) {
-    super(props)
-    this.refForm = React.createRef()
-    this.refPassword = React.createRef()
-  }
+  const refForm = React.createRef()
+  const refPassword = React.createRef()
```

Comment:
Does `createRef()` work inside the function componest, I remember it only wors inside the class component, see this: https://github.com/dt-fe/weekly/blob/v2/141.%E7%B2%BE%E8%AF%BB%E3%80%8AuseRef%20%E4%B8%8E%20createRef%20%E7%9A%84%E5%8C%BA%E5%88%AB%E3%80%8B.md

Maybe we need to use `useRef()`.

Reply:
Thanks! I use `useRef()` now, PTAL.

Reply:
Nice, the rest LGTM, please help merge the latest master branch code.